packages/flutter/lib/src/widgets/form.dart

```diff
@@ -242,7 +242,7 @@ class FormState extends State<Form> {
   int _generation = 0;
   bool _hasInteractedByUser = false;
   final Set<FormFieldState<dynamic>> _fields = <FormFieldState<dynamic>>{};
-
+  bool get _hasError => _fields.any((FormFieldState<dynamic> field) => field.hasError);
```

Comment:
nit: Instead of making this a property, could you move this logic to the `build` method? The property makes it look like this is an `O(1)` operation when it is actually `O(n)`. Moving this logic to the `build` method would make this more obvious.